src/Arcus.Observability.Tests.Unit/Telemetry/TelemetryTypeFilterTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Linq;
+using Arcus.Observability.Telemetry.Core;
+using Arcus.Observability.Telemetry.Serilog.Filters;
+using Bogus;
+using Serilog.Events;
+using Xunit;
+
+namespace Arcus.Observability.Tests.Unit.Telemetry
+{
+    [Trait("Category", "Unit")]
+    public class TelemetryTypeFilterTests
+    {
+        private readonly Faker _bogusGenerator = new Faker();
+        
+        [Fact]
+        public void LogEventAsTelemetry_FiltersInCorrectTelemetry_Succeeds()
+        {
+            // Arrange
+            DateTimeOffset timestamp = _bogusGenerator.Date.RecentOffset();
+            var level = _bogusGenerator.Random.Enum<LogEventLevel>();
+            var telemetryType = _bogusGenerator.Random.Enum<TelemetryType>();
+            var properties = new[]
+            {
+                new LogEventProperty(ContextProperties.General.TelemetryType, new ScalarValue(telemetryType))
+            };
+
+            var logEvent = new LogEvent(timestamp, level, exception: null, MessageTemplate.Empty, properties);
+            var filter = TelemetryTypeFilter.On(telemetryType, isTrackingEnabled: true);
+            
+            // Act
+            bool isEnabled = filter.IsEnabled(logEvent);
+            
+            // Assert
+            Assert.True(isEnabled);
+        }
+
+        [Fact]
+        public void LogEventAsTelemetry_FiltersOutCorrectTelemetry_Succeeds()
```

Comment:
I would add a test case like this one, but instead:
1. Tell it to filter a given telemetry type, for example dependency
2. Write a new log event with a different type
3. Entry is not filtered

Reply:
Yes, missed these ones. I'll add some theory for that. Thanks. Resolving.